modules/http-client/src/main/java/com/devonfw/module/httpclient/common/impl/AbstractSyncServiceHttpClient.java

```diff
@@ -0,0 +1,135 @@
+package com.devonfw.module.httpclient.common.impl;
+
+import java.io.IOException;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.net.http.HttpResponse.BodyHandlers;
+import java.util.function.Consumer;
+
+import com.devonfw.module.service.common.api.client.SyncServiceClient;
+import com.devonfw.module.service.common.api.client.async.ServiceClientInvocation;
+import com.devonfw.module.service.common.api.client.async.ServiceClientStub;
+import com.devonfw.module.service.common.base.client.AbstractSyncServiceClient;
+import com.devonfw.module.service.common.base.client.ServiceClientPerformanceLogger;
+
+/**
+ * Abstract base implementation of {@link SyncServiceClient} using Java HTTP client.
+ *
+ * @param <S> type of the {@link #get() service client}.
+ * @param <F> type of the owning {@link SyncServiceClientFactoryHttp factory}.
+ * @since 2020.08.001
+ */
+public abstract class AbstractSyncServiceHttpClient<S, F extends SyncServiceClientFactoryHttp>
+    extends AbstractSyncServiceClient<S> {
+
+  /** {@link ServiceHttpClient} to use. */
+  protected final ServiceHttpClient client;
+
+  /** The owning {@link SyncServiceClientFactoryHttp factory} which created this client. */
+  protected final F factory;
+
+  /**
+   * The constructor.
+   *
+   * @param proxy the {@link #get() service client}.
+   * @param stub the {@link ServiceClientStub}.
+   * @param httpClient the {@link ServiceHttpClient} to use.
+   * @param factory the owning {@link SyncServiceClientFactoryHttp factory}.
+   */
+  public AbstractSyncServiceHttpClient(S proxy, ServiceClientStub<S> stub, ServiceHttpClient httpClient, F factory) {
+
+    super(proxy, stub);
+    this.client = httpClient;
+    this.factory = factory;
+  }
+
+  @SuppressWarnings("null")
+  @Override
+  protected <R> void doCall(ServiceClientInvocation<S> invocation, Consumer<R> resultHandler) {
+
+    long startTime = System.nanoTime();
+    HttpRequest request = createRequest(invocation);
+    HttpResponse<String> future = null;
+    try {
+      future = this.client.getHttpClient().send(request, BodyHandlers.ofString());
+    } catch (IOException e) {
```

Comment:
Check for exception handling part in asynclient . do not add e.printStackTrace()